src/kdalgorithms.h

```diff
@@ -351,4 +370,59 @@ auto index_of_match(const Container &container, UnaryPredicate &&predicate)
     return result.has_result() ? std::distance(result.begin, result.iterator) : -1;
 }
 
+// -------------------- iota --------------------
+template <template <typename...> class Container = std::vector, typename Value>
+Container<Value> iota(Value initial, int count)
+{
+    Container<Value> result(count);
+    std::iota(std::begin(result), std::end(result), initial);
+    return result;
+}
+
+// -------------------- partition --------------------
+template <typename T>
+struct partition_result
+{
+    T in;
+    T out;
+};
+
+template <typename ResultContainer, typename Container, typename UnaryPredicate>
+auto partitioned(Container &&container, UnaryPredicate predicate)
+{
+    remove_cvref_t<ResultContainer> in;
```

Comment:
Maybe create a partition_result instance here to return it at the end of the function to trigger RVO instead of creating the collections and moving them into the result.